src/category_theory/monoidal/internal/functor_category.lean

```diff
@@ -0,0 +1,129 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.monoidal.internal
+import category_theory.monoidal.functor_category
+
+/-!
+# `Mon_ (C ⥤ D) ≌ C ⥤ Mon_ D`
+
+When `D` is a monoidal category,
+monoid objects in `C ⥤ D` are the same thing as functors from `C` into the monoid objects of `D`.
+
+This is formalised as:
+* `Mon_functor_category_equivalence : Mon_ (C ⥤ D) ≌ C ⥤ Mon_ D`
+
+The intended application is that as `Ring ≌ Mon_ Ab` (not yet constructed!),
+we have `presheaf Ring X ≌ presheaf (Mon_ Ab) X ≌ Mon_ (presheaf Ab X)`,
+and we can model a module over a presheaf of rings as a module object in `presheaf Ab X`.
+-/
+
+universes v₁ v₂ u₁ u₂
+
+open category_theory
+open category_theory.monoidal_category
+
+namespace category_theory.monoidal
+
+variables {C : Type u₁} [category.{v₁} C]
+variables {D : Type u₂} [category.{v₂} D] [monoidal_category.{v₂} D]
+
+namespace Mon_functor_category_equivalence
+
+/--
+Functor translating a monoid object in a functor category
+to a functor into the category of monoid objects.
+-/
+@[simps]
+def functor : Mon_ (C ⥤ D) ⥤ (C ⥤ Mon_ D) :=
+{ obj := λ A,
+  { obj := λ X,
+    { X := A.X.obj X,
+      one := A.one.app X,
+      mul := A.mul.app X,
+      one_mul' := congr_app A.one_mul X,
+      mul_one' := congr_app A.mul_one X,
+      mul_assoc' := congr_app A.mul_assoc X, },
```

Comment:
Do we want to keep this terminal comma (and the others below?)

Reply:
I'm pretty sure the current style guide leaves it up to preference. I like the trailing commas everywhere for some reason, but I'm happy to change if others don't.

Reply:
Fair enough!